### ISSUE TITLE
Loaded BufferGeometry no longer applies transforms to position array in r84

### ISSUE THREAD

**Issue by marquizzo** [mrdoob/three.js#11036]
##### Description of the problem 

In r83, a loaded `BufferGeometry` object would receive transforms, and apply these transform matrices to their "position" `Float32Array`.  However, in r84, these transform matrices are no longer applied. 

Here's a simple example where I apply a rotation matrix to duplicate a tree:
```
// Load tree geometry
var geomLoader = new THREE.BufferGeometryLoader();
geomLoader.load("meshes/tree.json", geomLoaded);

function geomLoaded(geometry){
    // Enlarge tree
    geometry.scale(2, 2, 2);

    // posArray is twice as big as the loaded vertices so it can hold two trees
    var vertCount = geometry.getAttribute("position").itemSize * 2;
    var posArray = new Float32Array(vertCount * 3);

    // Set top tree vertices to first half posArray
    posArray.set(geometry.getAttribute("position").array);

    // Rotate geometry and apply new vertices to second half of posArray
    geometry.rotateZ(Math.PI);
    posArray.set(geometry.getAttribute("position").array, posArray.length / 2);

    geometry.addAttribute("position", new THREE.BufferAttribute(posArray, 3));

    pointsObject = new THREE.Points(this.geometry, this.shaderMat);
    scene.add(pointsObject);
}
```

In r83, I got the desired effect of two trees in one `BufferGeometry` object:
![r83](https://cloud.githubusercontent.com/assets/7864858/24126635/447f364c-0d8c-11e7-9854-6bf4585da303.jpg)

In r84, I get the undesired effect of no scaling, and no rotating, leading to both trees stacking on top of each other:
![r84](https://cloud.githubusercontent.com/assets/7864858/24126637/47016ba6-0d8c-11e7-8c33-34849f4ed254.jpg)

Is this a bug, or an intentional change in behavior?

##### Three.js version
- [x] r84

##### Browser
- [x] All of them

##### OS

- [x] All of them

**Comment by mrdoob** [comment]
Could you create jsfiddles?

**Comment by jostschmithals** [comment]
I tried your code, and for me it works fine without any changes, if the loaded BufferGeometry is not indexed (apart from a mistake in the calculation of vertCount, where it must be `.count` instead of` .itemSize`, and two times `geometry` instead of `this.geometry`).

I tried it with an indexed Geometry too, and after doubling the index array (letting the second half pointing to the rotated vertices), this works fine for me, too.

Perhaps your geometry is indexed and maybe there was a change concerning indexed geometries in the meantime (I don't know)?

**Comment by marquizzo** [comment]
@jostschmithals My gosh, you're right. I forgot that I had exported (and perhaps manually edited) the JSON file a long time ago. It used to include the following syntax:

    "position":{
        "type":"Float32Array",
        "array":[0, 30, 29, /*...*/],
        "itemSize":90048
    }

I re-exported using the latest Blender exporter, and I got

    "position":{
        "type":"Float32Array",
        "array":[0, 30, 29, /*...*/],
        "itemSize":3
    }

So yes, you're right, to calculate the array length, I should've been using `geometry.count * 2`, instead of `geometry.itemSize * 2`. I'm guessing r83 used to be more forgiving, but glad to admit that it was an oversight on my part.
